File: python/FrogRiverOne.py
__author__ = 'kkamil'


# you can use print for debugging purposes, e.g.
# print "this is a debug message"

def solution(X, A):
    # write your code in Python 2.7
    leafs = [0]*X
    max = 0
    for num in range(len(A)):
        if leafs[A[num] - 1] == 0:
            leafs[A[num] - 1] = num
            max = num
    for leaf in leafs:
        if leaf == 0:
            return -1
    return max
    pass

File: cpp/Lesson2/FrogRiverOne.cpp
// you can use includes, for example:
// #include <algorithm>

// you can write to stdout for debugging purposes, e.g.
// cout << "this is a debug message" << endl;

int solution(int X, vector<int> &A) {
    // write your code in C++11
    vector<int> time(X,-1);
    
    for (unsigned int i = 0; i < A.size(); ++i)
    {
        if (time[A[i] - 1] == -1)
        {
            time[A[i] - 1] = i;
        }
    }
    
    int result = -1;
    
    for (auto it = time.begin(); it != time.end(); ++it)
    {
        if (*it == -1)
            return -1;
        if (*it > result)
            result = *it;
    }
    
    return result;
}

